test/src/test_derive.rs

```diff
@@ -160,3 +167,181 @@ fn test_derive_owned_to_variant() -> bool {
 
     ok
 }
+
+#[derive(gdnative::NativeClass)]
+#[inherit(Node)]
+struct FakeNativeClassWithPropertyHooks {
+    #[property(before_set = "Self::before_get_fake_callback")]
+    before_get_bool: bool,
+    #[property(after_set = "Self::after_get_fake_callback")]
+    after_get_bool: bool,
+    #[property(before_set = "Self::before_set_fake_callback")]
+    before_set_bool: bool,
+    #[property(after_set = "Self::after_set_fake_callback")]
+    after_set_bool: bool,
+
+    pub before_get_callback_calls: u32,
+    pub after_get_callback_calls: u32,
+    pub before_set_callback_calls: u32,
+    pub after_set_callback_calls: u32,
+}
```

Comment:
I think these names are a bit too verbose. Maybe instead of `FakeNativeClassWithPropertyHooks`, just make it `PropertyHooks`? Same with `Self::before_get_fake_callback` and `before_get_callback_calls` -- just make that `Self::before_get` and `before_get_calls`.